devicetypes/smartenit/iot8-z.src/iot8-z.groovy

```diff
@@ -0,0 +1,238 @@
+            } else if (eventDescMap?.clusterId == "000C" || eventDescMap?.cluster == "000C") {
+                long convertedValue = Long.parseLong(eventDescMap?.value, 16)
+                Float percentage = Float.intBitsToFloat(convertedValue.intValue())
+                percentage = (percentage / 1.60) * 100.0
+                def ceilingVal = Math.ceil(percentage)
+                if (ceilingVal > 100.0) {
+                    ceilingVal = 100.0
+                }
+                int intValue = (int) ceilingVal
+                
+                return createEvent(name: "inputValue", value: intValue)
+            }
+        } else {
+        	def childDevice = childDevices.find {
+				it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.sourceEndpoint}" || it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.endpoint}"
+			}
+            if (childDevice) {
+                if (eventDescMap?.clusterId == "000F" || eventDescMap?.cluster == "000F") {
+                	if (eventDescMap?.value == "00") {
+                        def map = createEvent(name: "contact", value: "open")
+                        childDevice.sendEvent(map)
+                    } else if (eventDescMap?.value == "01") {
+                        def map = createEvent(name: "contact", value: "closed")
+                        childDevice.sendEvent(map)
+                    }
+                } else if (eventDescMap?.clusterId == "000C" || eventDescMap?.cluster == "000C") {
+                    long convertedValue = Long.parseLong(eventDescMap?.value, 16)
+                    Float percentage = Float.intBitsToFloat(convertedValue.intValue())
+                    percentage = (percentage / 1.60) * 100.0
+                    def ceilingVal = Math.ceil(percentage)
+                    if (ceilingVal > 100.0) {
+                    	ceilingVal = 100.0
+                    }
+                    
+                    int intValue = (int) ceilingVal
+                    
+                    def map = createEvent(name: "inputValue", value: intValue)
+                    childDevice.sendEvent(map)
+            	}
+            }
+        }
+    }
+}
+
+// handle commands
+def on() {
+    zigbee.on()
+}
+
+def off() {
+    zigbee.off()
+}
+
+def childOn(String dni) {
+    def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x01, "", [destEndpoint: childEndpoint])
+}
+
+def childOff(String dni) {
+	def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x00, "", [destEndpoint: childEndpoint])
+
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	def refreshCommands = zigbee.onOffRefresh() 
+    def numberOfChildDevices = 8
+    
+	for(def endpoint : 2..numberOfChildDevices) {
+		refreshCommands += zigbee.readAttribute(zigbee.ONOFF_CLUSTER, 0x0000, [destEndpoint: endpoint])
+	}
+    for(def endpoint : 1..4) {
+    	refreshCommands += zigbee.readAttribute(0x000F, 0x0055, [destEndpoint: endpoint])
```

Comment:
Please change all "magic numbers" to constants, like:
`private getBATTERY_PERCENTAGE_REMAINING() { 0x0021 }`
to which later you can refer as
`BATTERY_PERCENTAGE_REMAINING`
(this particular example is from https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/leaksmart-water-sensor.src/leaksmart-water-sensor.groovy#L62 )